packages/aws-serverless/src/sdk.ts

```diff
@@ -320,15 +325,20 @@ export function wrapHandler<TEvent, TResult>(
         throw e;
       } finally {
         clearTimeout(timeoutWarningTimer);
-        span?.end();
+        if (span && span.isRecording()) {
+          span.end();
+        }
         await flush(options.flushTimeout).catch(e => {
           DEBUG_BUILD && logger.error(e);
         });
       }
       return rv;
     }
 
-    if (options.startTrace) {
+    // Only start a trace and root span if the handler is not already wrapped by Otel instrumentation
+    // Otherwise, we create two root spans (one from otel, one from our wrapper).
+    // If Otel instrumentation didn't work or was filtered by users, we still want to trace the handler.
+    if (options.startTrace && !isWrappedByOtel(handler)) {
```

Comment:
One thing to note here: Checking for `getActiveSpan` doesn't work because apparently, our patch is applied _before_ Otel's patch is applied. So I opted to check for the otel patch instead. I still think we should prefer Otel's span though because it contains more data than ours.